src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -571,9 +576,18 @@ abstract class ManagedStrategy(
             }
             return nextThread
         }
-        // otherwise exit if the thread switch is optional, or all threads are finished
-        if (!mustSwitch || threadScheduler.areAllThreadsFinished()) {
-           return iThread
+        // exit if all threads are finished
+        if (threadScheduler.areAllThreadsFinished()) {
+            return iThread
+        }
+        // try to unpark some parked thread
```

Comment:
Why should Lincheck unpark a parked thread?

Reply:
Suppose a thread is parked inside a silent section, and all other threads are blocked. 
Shouldn't the strategy then attempt to unpark it spuriously?

Reply:
Nevermind, reverted it.